www/js/types/diaryTypes.ts

```diff
@@ -118,13 +131,56 @@ export type UserInputEntry = {
   key?: string;
 };
 
-export type LocalDt = {
-  minute: number;
-  hour: number;
-  second: number;
-  day: number;
-  weekday: number;
-  month: number;
-  year: number;
-  timezone: string;
+export type Location = {
+  speed: number;
+  heading: number;
+  local_dt: LocalDt;
+  idx: number;
+  section: ObjectId;
+  longitude: number;
+  latitude: number;
+  fmt_time: string; // ISO
+  mode: number;
+  loc: LocationCoord;
+  ts: number; // Unix
+  altitude: number;
+  distance: number;
+};
+
+// used in readAllCompositeTrips
+export type SectionData = {
+  end_ts: number; // Unix time, e.x. 1696352498.804
+  end_loc: LocationCoord;
+  start_fmt_time: string; // ISO time
+  end_fmt_time: string;
+  trip_id: ObjectId;
+  sensed_mode: number;
+  source: string; // e.x., "SmoothedHighConfidenceMotion"
+  start_ts: number; // Unix
+  start_loc: LocationCoord;
+  cleaned_section: ObjectId;
+  start_local_dt: LocalDt;
+  end_local_dt: LocalDt;
+  sensed_mode_str: string; //e.x., "CAR"
+  duration: number;
+  distance: number;
+};
+
+export type GjFeature = {
+  type: string;
+  geometry: LocationCoord;
+  properties?: { featureType: string }; // if geometry.coordinates.length == 1, property
+  style?: { color: string }; // otherwise, style (which is a hexcode)
+};
+
+export type GeoJSON = {
+  data: {
+    id: string;
+    type: string;
+    features: GjFeature[];
+    properties: {
+      start_ts: number;
+      end_ts: number;
+    };
+  };
```

Comment:
Thank you for doing this typing work — but I think we should be able to grab these types from a package. That way we can ensure our type defs align with the official GeoJSON spec.
I did some digging and I think Leaflet's types package includes the geojson type package too.

So, can you `npm i -D @types/leaflet` (updating `package.serve.json` and `package.cordovabuild.json` accordingly).
Then see if you can `import { Feature, FeatureCollection, GeoJsonObject } from 'geojson'` and use those types instead of declaring them ourselves

If you see something we're doing here that doesn't line up with the GeoJSON spec, feel free to fix it!

Reply:
I think we may end up with something like:
```ts
export type GeoJSON = {
  id: string, // the official spec doesn't have an 'id' field so we declare it ourselves
  data: GeoJsonObject, // everything that *is* in the official spec goes inside 'data'
}

Reply:
Actually, maybe something like this is better:
```ts
export type GeoJSON = GeoJsonObject & { id: string };
export type GjFeature = Feature & { style?: { color: string} };
```

Reply:
Try it and see what works!